app/src/main/java/com/islamey/hamza/wazaker/ui/HomeFragment/HijriViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package com.islamey.hamza.wazaker.ui.HomeFragment
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.islamey.hamza.wazaker.domain.Models.HijriDateResponse
+import com.islamey.hamza.wazaker.domain.repo.HijriRepo
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+sealed class DataState<out T> {
+    object Idle : DataState<Nothing>()
+    object Loading : DataState<Nothing>()
+    data class Success<T>(val data: T) : DataState<T>()
+    data class Error(val message: String) : DataState<Nothing>()
+}
```

Comment:
I see you use generics here, so I think to move it outside fragment and some common place or modeling layer